lib/datautils/owid/datautils/dataframes.py

```diff
@@ -537,7 +544,7 @@ def concatenate(objs: List[pd.DataFrame], **kwargs: Any) -> pd.DataFrame:
         uc = union_categoricals([df[col] for df in objs], ignore_order=ignore_order)
         # Change to union category for all dataframes
         for df in objs:
-            df.loc[:, col] = pd.Categorical(df[col].values, categories=uc.categories)
+            df[col] = pd.Categorical(df[col].values, categories=uc.categories)
```

Comment:
This is equivalent, except that with `df[col]` you are just setting a new column, while with `df.loc[:, col]` values are being set inplace into the existing column. And that is again a case of where a setitem operation will try to preserve the original dtype when setting values inplace (i.e. not when replacing an entire column. And of course with a `:` slice this is a bit ambiguous, as your intention is to replace the entire column)

Reply:
Thanks, it was always a bit unclear to us what's the difference between the two. We use `.loc[:, col]` because it plays better with our subclassed dataframe class `Table` (type checker is confused by `df[col]` and thinks we're casting it back to DataFrame)